api/lib/controllers/meeting.js

```diff
@@ -314,18 +365,34 @@ module.exports = {
             },
           },
         },
-        { $unwind: "$meetings" },
-        { $replaceRoot: { newRoot: "$meetings" } },
-        { $sort: { date: -1 } },
-        { $skip: parseInt(skip) },
-        { $limit: parseInt(limit) },
+
+        {
+          $project: {
+            meetings: {
+              $sortArray: {
+                input: '$meetings',
+                sortBy: { date: -1 }
+              }
+            },
+            count: { $size: '$meetings' }
+          }
+        },
```

Comment:
Right now the owners filter only shows the owners in the slice that the user is looking at. To fix that we can take the unique owners out of the `$meetings` array and place them into a _set_ here before we `$slice` the meetings. We will still be missing owners that are filtered out due to other filters (since those filters will have already been applied), but we can get around that by preserving the owners from the initial unfiltered query on page load.